src/arrays/arrays_with_methods/hard/q12/MainTest.java

```diff
@@ -0,0 +1,62 @@
+package arrays.arrays_with_methods.hard.q12;
+
+import global.BaseTest;
+import global.MethodTest;
+import global.exceptions.InvalidClauseException;
+import global.tools.CustomAssertions;
+import global.tools.TestOption;
+import global.variables.Clause;
+import global.variables.clauses.IntegerLiteral;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.PlaceHolder;
+import global.variables.clauses.StringLiteral;
+import global.variables.wrappers.Optional;
+import org.junit.jupiter.api.RepeatedTest;
+import org.junit.jupiter.api.RepetitionInfo;
+
+public class MainTest extends BaseTest {
+    // Java
+
+    int[] factorials = {1, 2, 6, 24, 120, 720, 5040, 40320, 362880, 3628800, 39916800, 479001600, 1932053504, 1278945280, 2004310016};
+
+    public Clause[] testSentence() {
```

Comment:
I think I asked for arrayUtil in this case by accident. I meant that we should maybe have a larger test case if possible so we can make sure that they are calculating factorial and not just using a switch or something.